src/typed-error.ts

```diff
@@ -19,6 +19,7 @@ export class TypedError extends Error {
 	public name: string;
 	public message: string;
 	public stack: string = '';
+	public payload: any | null;
```

Comment:
How do you envision people setting this property? It looks like you're imagining something like:

```ts
const error = new SomethingThatExtendsTypedError();
error.payload = { ... };
throw error;
```

That's a bit awkward. Could we make this an optional constructor parameter? Then subclasses could pass payloads to `super` if they like, and that'd give:

```ts
throw new SomethingThatExtendsTypedError({ ... });
```

That would mean you'd need to update each subclass to actively pass the parameter up though... Hmm. 

Alternatively, we could implement a method that sets it and then returns the whole error? Bit clever for its own good, so I'd be curious to hear what @Page- things, but that'd look like:

```ts
throw new SomethingThatExtendsTypedError().withPayload({ ... });
```

(where with payload returns the error itself again, effectively chaining)

For that, we could implement a method here, and all errors extending typed error would get it automatically, so you could add payloads to anything.